### ISSUE TITLE
New updates caused 4 new errors

### ISSUE THREAD

**Issue by epiqueras** [micwallace/wallacepos#74]
I pulled the latest changes to see if they would fix the SQL GROUP_BY error.
The error is gone, but now I have four new ones.

On startup:

    WARNING: fopen(/app/docs/logs/wpos_log_17-01-27.txt): failed to open stream: No such file or
    directory /app/library/wpos/models/Logger.php on line 44
    WARNING: fwrite() expects parameter 1 to be resource, boolean given
    /app/library/wpos/models/Logger.php on line 46
    WARNING: fclose() expects parameter 1 to be resource, boolean given
    /app/library/wpos/models/Logger.php on line 48

On loading Dashboard, Realtime, Reports, or Graph:

    SQLSTATE[42S22]: Column not found: 1054 Unknown column 'cost' in 'field list'

On loading Realtime:

    Socket authentication failed! failed: Invalid frame header

On loading POS:

    Application Cache Error event: Manifest fetch failed (500)

**Comment by micwallace** [comment]
You updated the code but didn't run the application updater. Go to /installer to update.

**Comment by epiqueras** [comment]
Oops

**Comment by epiqueras** [comment]
I just tried running it. It gives me a mixed content error.

**Comment by micwallace** [comment]
Ok that needs to be fixed, however you can access that url directly using https and it should do the upgrade.

**Comment by epiqueras** [comment]
I get this error when I try to access the link directly:

    Backing up database...
    <br />
    <b>Warning</b>:  fopen(/app/docs/backup/dbbackup-2017-01-27_05-11-38.sql): failed to open
    stream: No such file or directory in <b>/app/library/ifsnop/mysqldump
    php/src/Ifsnop/Mysqldump/Mysqldump.php</b> on line <b>1089</b><br />
    <br />
    <b>Fatal error</b>:  Uncaught exception 'Exception' with message 'Output file is not writable'
    in /app/library/ifsnop/mysqldump-php/src/Ifsnop/Mysqldump/Mysqldump.php:1091
    Stack trace:
    #0 /app/library/ifsnop/mysqldump-php/src/Ifsnop/Mysqldump/Mysqldump.php(320):
    Ifsnop\Mysqldump\CompressNone-&gt;open('/app/docs/backu...')
    #1 /app/library/wpos/models/WposAdminUtilities.php(257): Ifsnop\Mysqldump\Mysqldump
    &gt;start('/app/docs/backu...')
    #2 /app/library/wpos/models/DbUpdater.php(149): WposAdminUtilities::backUpDatabase(false)
    #3 /app/installer/index.php(180): DbUpdater-&gt;upgrade(NULL)
    #4 {main}
    thrown in <b>/app/library/ifsnop/mysqldump-php/src/Ifsnop/Mysqldump/Mysqldump.php</b>
    on line <b>1091</b><br />

**Comment by micwallace** [comment]
Please read the error, it provides useful information. docs/backup folder is not writable. Check permissions.

**Comment by micwallace** [comment]
Perhaps I was a bit hasty with my last reply, I noticed that you are using dokku. Unfortunately since you did not have persistent storage enabled on the docs directory, those files were lost during the deployment.

To prevent this issue next time, run the following:

`dokku storage:mount %APP_NAME% /var/lib/dokku/data/storage/%APP_NAME%:/app/docs`

I have updated the readme with this information.

To fix the issue in the current container, attach to the container and copy the directory docs-template to docs.
